v2/devdoc/log_context_property_type_ascii_char_ptr_requirements.md

```diff
@@ -0,0 +1,95 @@
+# `log_context_property_type_ascii_char_ptr` requirements
+
+`log_context_property_type_ascii_char_ptr` implements the code needed for handling an ASCII string.
+
+## Exposed API
+
+```
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(ascii_char_ptr)(void* dst_value, const char* src_value);
+int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(ascii_char_ptr)(const char* format, ...);
+
+extern const LOG_CONTEXT_PROPERTY_TYPE_IF LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr);
+
+```
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(void* property_value, char* buffer, size_t buffer_length);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` produces the string representation of an `ascii_char_ptr`.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_001: [** If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_017: [** If `buffer` is `NULL` and `buffer_length` is greater than 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_002: [** If `buffer` is `NULL` and `buffer_length` is 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall return the length of the string pointed to by `property_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_003: [** Otherwise, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall copy the string pointed to by `property_value` to `buffer` by using `snprintf` with `buffer`, `buffer_length` and format string `%s` and pass in the values list the `const char*` value pointed to be `property_value`. **]**
+
+Note: the copying of the string can be somewhat optimized by not using `snprintf`, this will be considered in a separate task.
```

Comment:
>in a separate task [](http://example.com/codeflow?start=107&length=18)

Why not just do the memcpy right here and now?

Reply:
Hm, maybe mocking and testing would be more complicated. I guess I will see.

Reply:
I did not want to optimize yet, simplest was to throw it for now to snprintf. This would be for console logging only anyway.